experiments/scripts/run_tune.py

```diff
@@ -111,12 +112,51 @@ def run(config_path: str, config_overrides: Optional[Dict] = None) -> None:
         f"Total Parameters: {model.num_parameters()} Trainable Parameters: {total_trainables}",
     )
 
-    dataset = datasets.load_from_disk(config.data.path)
-    split_dataset = dataset.train_test_split(
-        test_size=config.data.validation_size, shuffle=False
-    )
-    data_collator = DataCollatorForLanguageModeling(tokenizer, mlm=False)
+    if isinstance(config.data.path, list):
+        # set validation size per dataset or constant
+        validation_sizes = (
+            [config.data.validation_size] * len(config.data.path)
+            if isinstance(config.data.validation_size, float)
+            else config.data.validation_size
+        )
+        assert len(config.data.path) == len(
+            validation_sizes
+        ), "you must specify an equal number of datasets and validation sizes"
+
+        # collect all datasets
+        data_sources = {
+            source: load_and_split(source, val_size)
+            for source, val_size in zip(config.data.path, validation_sizes)
+        }
+        train_ds = [d["train"] for d in data_sources.values()]
+        if config.data.interleave_probs:
+            # interleaving with specific probabilities and stopping criterion
+            assert (
+                config.data.interleave_probs and config.data.sampling_criterion
```

Comment:
Sorry if I just totally missed where it says this, but I can see that data.sampling_criterion must be a string, but it's not clear to me what values string it can take?

Reply:
Ye, they currently can only be one of two options (which are now added as type hints)